GPU/GPUTracking/Global/GPUChainTracking.cxx

```diff
@@ -870,171 +870,185 @@ int GPUChainTracking::RunTPCClusterizer()
+        SetupGPUProcessor(&clusterer, false);
+        clusterer.mPmemory->counters.nPeaks = clusterer.mPmemory->counters.nClusters = 0;
+        clusterer.mPmemory->fragment = fragment;
+
+        if (propagateMCLabels) {
+          clusterer.PrepareMC();
+          clusterer.mPinputLabels = digitsMC->v[iSlice];
+          // TODO: Why is the number of header entries in truth container
+          // sometimes larger than the number of digits?
+          assert(clusterer.mPinputLabels->getIndexedSize() >= mIOPtrs.tpcPackedDigits->nTPCDigits[iSlice]);
+        }
 
-      unsigned int nPagesTotal = 0;
-      if (mIOPtrs.tpcZS) {
-        for (unsigned int j = 0; j < GPUTrackingInOutZS::NENDPOINTS; j++) {
-          for (unsigned int k = 0; k < mIOPtrs.tpcZS->slice[iSlice].count[j]; k++) {
-            nPagesTotal += mIOPtrs.tpcZS->slice[iSlice].nZSPtr[j][k];
+        unsigned int nPagesTotal = 0;
+        if (mIOPtrs.tpcZS) {
+          for (unsigned int j = 0; j < GPUTrackingInOutZS::NENDPOINTS; j++) {
+            for (unsigned int k = 0; k < mIOPtrs.tpcZS->slice[iSlice].count[j]; k++) {
+              nPagesTotal += mIOPtrs.tpcZS->slice[iSlice].nZSPtr[j][k];
+            }
           }
         }
-      } else {
-        auto* inDigits = mIOPtrs.tpcPackedDigits;
-        size_t numDigits = inDigits->nTPCDigits[iSlice];
-        clusterer.mPmemory->counters.nDigits = numDigits;
-        if (doGPU) {
-          GPUMemCpy(RecoStep::TPCClusterFinding, clustererShadow.mPdigits, inDigits->tpcDigits[iSlice], sizeof(clustererShadow.mPdigits[0]) * numDigits, lane, true);
-        } else {
-          clustererShadow.mPdigits = const_cast<deprecated::Digit*>(inDigits->tpcDigits[iSlice]); // TODO: Needs fixing (invalid const cast)
+        if (not mIOPtrs.tpcZS || propagateMCLabels) {
+          auto* inDigits = mIOPtrs.tpcPackedDigits;
+          size_t numDigits = inDigits->nTPCDigits[iSlice];
+          clusterer.mPmemory->counters.nDigits = numDigits;
+          if (doGPU) {
+            GPUMemCpy(RecoStep::TPCClusterFinding, clustererShadow.mPdigits, inDigits->tpcDigits[iSlice], sizeof(clustererShadow.mPdigits[0]) * numDigits, lane, true);
+          } else {
+            clustererShadow.mPdigits = const_cast<o2::tpc::Digit*>(inDigits->tpcDigits[iSlice]); // TODO: Needs fixing (invalid const cast)
+          }
+        }
+        TransferMemoryResourceLinkToGPU(RecoStep::TPCClusterFinding, clusterer.mMemoryId, lane);
+
+        // These buffers only have to be cleared once entirely. The 'resetMaps' kernel
+        // takes care of subsequent clean ups.
+        if (iSliceBase == 0) {
+          using ChargeMapType = decltype(*clustererShadow.mPchargeMap);
+          using PeakMapType = decltype(*clustererShadow.mPpeakMap);
+          runKernel<GPUMemClean16>({BlockCount(), ThreadCount(), lane, RecoStep::TPCClusterFinding}, krnlRunRangeNone, {}, clustererShadow.mPchargeMap, TPCMapMemoryLayout<ChargeMapType>::items() * sizeof(ChargeMapType));
```

Comment:
Could you please stick to using `GetGridBlkStep(BlockCount(), lane, RecoStep::TPCClusterFinding)`

Reply:
Fixed. This also fixes the performance regression.